server/src/main/java/org/elasticsearch/rest/RestRequest.java

```diff
@@ -216,14 +240,20 @@ final void setXContentType(XContentType xContentType) {
         this.xContentType.set(xContentType);
     }
 
-    @Nullable
-    public SocketAddress getRemoteAddress() {
-        return null;
+    public InetSocketAddress getRemoteAddress() {
```

Comment:
would it make sense to let the consumer just call `getHttpChannel` and then get from it what they need instead of having these delegates?